quickstarts/microsoft.compute/vmss-windows-autoscale/main.bicep

```diff
@@ -0,0 +1,243 @@
+var loadBalancerName = '${namingInfix}lb'
+var natPoolName = '${namingInfix}natpool'
+var bePoolName = '${namingInfix}bepool'
+
+var natStartPort = 50000
+var natEndPort = 50119
+var natBackendPort = 3389
+var nicName = '${namingInfix}nic'
+var ipConfigName = '${namingInfix}ipconfig'
+
+var osType = {
+  publisher: 'MicrosoftWindowsServer'
+  offer: 'WindowsServer'
+  sku: windowsOSVersion
+  version: 'latest'
+}
+var imageReference = osType
+
+resource virtualNetwork 'Microsoft.Network/virtualNetworks@2021-02-01' = {
+  name: virtualNetworkName
+  location: location
+  properties: {
+    addressSpace: {
+      addressPrefixes: [
+        addressPrefix
+      ]
+    }
+    subnets: [
+      {
+        name: subnetName
+        properties: {
+          addressPrefix: subnetPrefix
+        }
+      }
+    ]
+  }
+}
+
+resource publicIP 'Microsoft.Network/publicIPAddresses@2021-02-01' = {
+  name: publicIPAddressName
+  location: location
+  properties: {
+    publicIPAllocationMethod: 'Dynamic'
+    dnsSettings: {
+      domainNameLabel: longNamingInfix
+    }
+  }
+}
+
+resource loadBalancer 'Microsoft.Network/loadBalancers@2021-02-01' = {
+  name: loadBalancerName
+  location: location
+  properties: {
+    frontendIPConfigurations: [
+      {
+        name: 'LoadBalancerFrontEnd'
+        properties: {
+          publicIPAddress: {
+            id: publicIP.id
+          }
+        }
+      }
+    ]
+    backendAddressPools: [
+      {
+        name: bePoolName
+      }
+    ]
+    inboundNatPools: [
+      {
+        name: natPoolName
+        properties: {
+          frontendIPConfiguration: {
+            id: resourceId('Microsoft.Network/loadBalancers/frontendIPConfigurations', loadBalancerName, 'loadBalancerFrontEnd')
+          }
+          protocol: 'Tcp'
+          frontendPortRangeStart: natStartPort
+          frontendPortRangeEnd: natEndPort
+          backendPort: natBackendPort
+        }
+      }
+    ]
+  }
+}
+
+resource vmss 'Microsoft.Compute/virtualMachineScaleSets@2021-03-01' = {
+  name: vmssName
+  location: location
+  sku: {
+    name: vmSku
+    tier: 'Standard'
+    capacity: instanceCount
+  }
+  properties: {
+    overprovision: true
+    upgradePolicy: {
+      mode: 'Manual'
+    }
+    virtualMachineProfile: {
+      storageProfile: {
+        osDisk: {
+          createOption: 'FromImage'
+          caching: 'ReadWrite'
+        }
+        imageReference: imageReference
+      }
+      osProfile: {
+        computerNamePrefix: namingInfix
+        adminUsername: adminUsername
+        adminPassword: adminPassword
+      }
+      networkProfile: {
+        networkInterfaceConfigurations: [
+          {
+            name: nicName
+            properties: {
+              primary: true
+              ipConfigurations: [
+                {
+                  name: ipConfigName
+                  properties: {
+                    subnet: {
+                      id: '${virtualNetwork.id}/subnets/${subnetName}'
```

Comment:
seems like we should be consistent with the construction of resourceIds that bicep doesn't handle for us... either use the resourceId() function or string interp...

resourceId() probably makes the most sense, consistent, familiar and intentional.

Reply:
Or better, maybe the subnet a resource so we can call subnet.id

Reply:
that works too - though more of a refactor and causes downtime on redeploy.  We also have other resources that don't have children but still have resourceIds.